runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java

```diff
@@ -1151,6 +1123,16 @@ private void drawFrame()
 			gl.glUniform1f(uniBrightness, (float) textureProvider.getBrightness());
 			gl.glUniform1f(uniSmoothBanding, config.smoothBanding() ? 0f : 1f);
 
+			// Calculate projection matrix
+			Matrix4 matrix = new Matrix4();
+			matrix.scale(0.04f / viewportWidth, 0.04f / viewportHeight, 1);
+			matrix.makeFrustum(-1, 1, -1, 1, 50f, 10000);
```

Comment:
this can cause far clipping in some cases, 20k should be safe (scene is 128*104 units in each dimension)